packages/@react-spectrum/dialog/src/DialogTrigger.tsx

```diff
@@ -66,13 +71,13 @@ export function DialogTrigger(props: SpectrumDialogTriggerProps) {
       case 'modal':
         return (
           <Modal isOpen={isOpen} onClose={onClose}>
-            {content}
+            {typeof content === 'function' ? content(onClose) : content}
```

Comment:
I'm not a fan of asking the people who create modals to write `(onClose) => {<content><Button onPress={close}>Close</Button></content>}`. It feels like we're asking them to write jsx that isn't just elements.

The first idea I thought of was a modal context to pass the onClose to button children which is also problematic, but it doesn't make the end user jump through any weird hoops.

I'm concerned that this will be a support issue of us having to train everyone to write things this way.

Reply:
This is a pretty common React pattern called 'Function as children' so it shouldn't be hard to understand.
This should also be documentable via our prop tables.
There isn't really a great way to do this via context without creating <DialogButton or something equivalent that can pick up a different context than the slots and knows which of the three (four) possible buttons it is.

We went through many options before arriving at this one. I think Devon might have it written down.